<h1>Importando os Dados</h1>

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

DataFrame = pd.read_csv('kc_house_data.csv', encoding='UTF-8', sep=";")
DataFrame.sample(10)

id        date     price  bedrooms  bathrooms  sqft_living  \
19843  7853321180  2014-12-22  465000.0         5       2.50         2550   
13074  7129304225  2015-01-08  305000.0         4       2.25         2340   
500    7852170140  2015-05-10  650000.0         4       2.50         3180   
19314  2720069019  2014-11-03  316000.0         3       1.75         1120   
3396   2721049059  2014-05-28  225000.0         3       2.00         2030   
16132  1246700050  2014-06-12  370000.0         2       1.00         1220   
7577     50300220  2015-02-12  363000.0         4       2.50         2180   
15921  7831800460  2014-05-02  235000.0         2       1.00         1210   
19200   123039147  2015-03-19  464950.0         3       2.00         2190   
2233    255400070  2014-07-01  875000.0         5       3.50         3840   

       sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
19843      6405     2.0           0     0     ...          7        2550   
13074      5250     2.0           0     0     ...          7        2340   
500        5438     2.0           0     0     ...          9        3180   
19314     98445     1.5           0     2     ...          7        1120   
3396      24829     1.0           0     0     ...          7        1220   
16132     17172     1.0           0     0     ...          6        1220   
7577       9281     2.0           0     0     ...          8        2180   
15921      9400     1.0           0     0     ...          6        1210   
19200     19800     1.0           0     0     ...          7        2190   
2233       8279     2.0           0     0     ...          9        3840   

       sqft_basement  yr_built  yr_renovated  zipcode      lat       long  \
19843              0      2008             0    98065  47,5191  -121,8690   
13074              0      1965             0    98118  47,5183  -122,2660   
500                0      2003             0    98065  47,5416  -121,8640   
19314              0      1917             0    98022  47,1853  -122,0170   
3396             810      1979             0    98001  47,2718  -122,2910   
16132              0      1947             0    98033  47,6934  -122,1630   
7577               0      2004             0    98042  47,3673  -122,0720   
15921              0      1949             0    98106  47,5342  -122,3600   
19200              0      1994             0    98146  47,5106  -122,3650   
2233               0      2001             0    98074  47,6039  -122,0590   

      sqft_living15  sqft_lot15  
19843          2190        5900  
13074          1540        5250  
500            3030        5335  
19314          1620       34200  
3396           1980       15204  
16132          1510       12915  
7577           2520        9520  
15921          1580        6026  
19200          1640        9719  
2233           3570        8279  

[10 rows x 21 columns]

In [34]:
!pip install seaborn --upgrade #or also !pip install seaborn==0.9.0

Invalid requirement: '#or'
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\packaging\requirements.py", line 92, in __init__
    req = REQUIREMENT.parseString(requirement_string)
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\pyparsing.py", line 1617, in parseString
    raise exc
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\pyparsing.py", line 1607, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\pyparsing.py", line 3376, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "C:\ProgramData\Anaconda3\lib\site-packages\pip\_vendor\pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File

In [14]:
print(DataFrame.price.min())
DataFrame.price.max()


75000.0


316875000000.0

<h3>A empresa, House Rocket, deseja obter infromações e o direcionamento de mercado de casas na cidade de King County, USA, para isto ela levantou as seguintes questões: </h3>
<h3> 
    <ol>
    <li>Quais casas o CEO da House Rocket deveria comprar e por qual preço de compra?</li>  <br>   
    <li>Uma vez a casa em posse da empresa, qual o melhor momento para vendê-las e qual seria o preço da venda?</li>   <br>  
    <li> A House Rocket deveria fazer uma reforma para aumentar o preço da venda? Quais seriam as sugestões de mudanças? Qual o incremento no preço dado por cada opção de reforma?</li>       
    </ol>
</h3>

<h6>Informações</h6>

sqft_basement = tamanho do porão

sqft_above = tamanho segundo andar

yr_built = ano de construção

yr_renovated = ano da reforma

# Observando casas reformadas para calcular sua valorização

In [3]:
Renovated = DataFrame[DataFrame['yr_renovated']!=0]
Unrenovated = DataFrame[DataFrame['yr_renovated']==0]

In [4]:
DataFrame.loc[0,:]

id               7129300520
date             2014-10-13
price                221900
bedrooms                  3
bathrooms                 1
sqft_living            1180
sqft_lot               5650
floors                    1
waterfront                0
view                      0
condition                 3
grade                     7
sqft_above             1180
sqft_basement             0
yr_built               1955
yr_renovated              0
zipcode               98178
lat                 47,5112
long              -122,2570
sqft_living15          1340
sqft_lot15             5650
Name: 0, dtype: object

In [5]:
Ren = Renovated['price'].mean()
Ren

1446211405.583151

In [6]:
Unr = Unrenovated['price'].mean()
Unr

529524767.00782645

In [7]:
print(Ren - Unr)

916686638.5753247


<h3> Pelos passos acima podemos perceber que na média as casas reformadas são mais valorizadas que as casas sem reforma.
    
Sendo assim seria interessante o CEO reformar a casa antes de Vendê-la.
</h3>

<h3> Seguindo esta lógica, o recomendado seria o CEO comprar uma casa mais barata para reformar e revender.<br>
    
Primeiramente será analisada a casa que esteja entre as mais antigas e será verificado se esta resulta na casa mais barata, ou em um preço baixo.</h3>



# Buscando a casa mais antiga para comparar com o preço da mais barata

In [8]:
Cheaper = Unrenovated[Unrenovated['yr_built'] == Unrenovated['yr_built'].min()]
CheapestElderHouse = Cheaper[Cheaper['price'] == Cheaper['price'].min()]
print(CheapestElderHouse)
CheapestElderHouse['condition']

               id        date     price  bedrooms  bathrooms  sqft_living  \
14001  2354300456  2015-03-11  130000.0         2        1.0          600   

       sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
14001      1500     1.0           0     0     ...          4         600   

       sqft_basement  yr_built  yr_renovated  zipcode      lat       long  \
14001              0      1900             0    98027  47,5289  -122,0330   

      sqft_living15  sqft_lot15  
14001          1130        6000  

[1 rows x 21 columns]


14001    4
Name: condition, dtype: int64

In [9]:
CheapestHouse = Unrenovated[Unrenovated['price'] == Unrenovated['price'].min()]
print(CheapestHouse)
CheapestHouse['condition']

              id        date    price  bedrooms  bathrooms  sqft_living  \
1149  3421079032  2015-02-17  75000.0         1        0.0          670   

      sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
1149     43377     1.0           0     0     ...          3         670   

      sqft_basement  yr_built  yr_renovated  zipcode      lat       long  \
1149              0      1966             0    98022  47,2638  -121,9060   

     sqft_living15  sqft_lot15  
1149          1160       42882  

[1 rows x 21 columns]


1149    3
Name: condition, dtype: int64

<h3> Diferente do esperado, nos resultados acima foi encontrada a casa mais antiga não renovada no valor de 130 mil, e também a mais barata encontrada no banco de dados, no valor de 75 mil.
    
    Analisando os preços pode-se dizer que não existe motivos evidentes para o CEO comprar a casa mais antiga, pois ambas casas tem o espaço de construção parecido (sqft_living) e apesar da casa de 75 mil estar numa condição pior e ter menos banheiros e quartos, isto pode ser solucionado na reforma, valorizando a casa, além de pagar 55 mil a menos. <br>
    
Apenas faria sentido comprar a casa antiga se fosse para uso imediato, sem pretensões de lucro, pois ela dispõe de uma melhor condição de moradia e recursos que a casa de 75 mil.
    
<br>
Sendo assim seria indicado que o CEO comprasse casas baratas no geral, entre 75 mil e 100mil, não importando muito com seu tempo de construção, uma vez que este não indica uma diminuição nos preços e nem com suas condições e itens como banheiros e quartos, uma vez que estes itens poderiam ser construidos numa reforma desde que o "sqft_lot" o possibilite.
</h3>

# Entendendo se o ZipCode impacta nos preços das casas

In [10]:
df = DataFrame.groupby('zipcode').mean()
df2 = DataFrame.groupby('zipcode').median()
df3 = DataFrame.groupby('zipcode').std()
ExpensiveZip = df[df['price'] == df['price'].max()]
print('ZipCode com casas mais caras em média', ExpensiveZip.index.values[0], 'valor médio', df['price'].max())

ZipCode com casas mais caras em média 98039 valor médio 14294837700.0


In [11]:
CheapZip = df[df['price'] == df['price'].min()]
CheapZipmedian = df2[df2['price'] == df2['price'].min()]
CheapZipstd = df3[df3['price'] == df3['price'].min()]
generalMean = DataFrame['price'].mean()
generalMedian = DataFrame['price'].median()
generalstd = DataFrame['price'].std()
print('ZipCode com as casas de médias mais baratas fica no ',CheapZip.index.values,
      'valor médio de: %i' %(df['price'].min()))
print('média geral é de: %i' %(generalMean))
print('ZipCode com as casas de menor mediana fica no ',CheapZipmedian.index.values,
      'valor médio de: %i' %(df2['price'].min()))
print('média geral é de: %i' %(generalMedian))
print('ZipCode com as casas com menor desvio padrão fica no ',CheapZipstd.index.values,
      'valor médio de: %i' %(df3['price'].min()))
print('média geral é de: %i' %(generalstd))

ZipCode com as casas de médias mais baratas fica no  [98002] valor médio de: 234284
média geral é de: 568290860
ZipCode com as casas de menor mediana fica no  [98002 98168] valor médio de: 235000
média geral é de: 450000
ZipCode com as casas com menor desvio padrão fica no  [98002] valor médio de: 52041
média geral é de: 8516008693


<h3> Outra variável que pode impactar o preço da casa é a localização, podemos análisar isso através da coluna "zipcode", e como visto anteriormente nos resultados seria interessante evitar o zipcode 98002, já que as medidas estatísticas apontam para um baixo valor das casas além de uma taxa de preço com pequenas variações, ou seja, muito provavelmente a localização tem grande impacto no valor da casa.<br>

<br>
Sabendo o valor médio das casas em King County, é possível afirmar que a reforma é uma boa estratégia, uma vez que o valor médio das casas reformadas, que é de 1.446.211.405, é maior que o valor médio das casas em geral e das casas não reformadas, como visto anteriormente.
</h3>

In [19]:
ID_Duplo = DataFrame[DataFrame['id'].duplicated(keep = False)]

ID_Duplo_Length = ID_Duplo.shape[0]
#CALCULANDO A VALORIZAÇÃO/DESVALORIZAÇÃO PELO TEMPO ENTRE OS IDS IGUAIS E SALVANDO OS VALORES EM meanProfit
meanProfit = []
for i in range(0, ID_Duplo_Length, 2):
    if(i+1 < ID_Duplo_Length):
        locActual = ID_Duplo.iloc[i,:]
        locProx = ID_Duplo.iloc[i+1,:]
        if(locActual.id == locProx.id):
            if locActual['date'] > locProx['date']:
                meanProfit.append(locActual['price']/locProx['price'])
            else:
                meanProfit.append(locProx['price']/locActual['price'])
        else:
            meanProfit.append(0)
    else:
        meanProfit.append(0)
        
#montando DataFrame com a valorização ou desvalorização dos imóveis
df_meanProfit = pd.DataFrame(meanProfit, columns = ['percentage'])    
df_meanProfit = df_meanProfit[df_meanProfit['percentage'] != 0]
#Passando de lista para DataFrame e retirando valores zerados

df_meanProfit.sort_values(by=['percentage'])

percentage
59      0.000099
1       0.001012
37      0.001024
118     0.945946
42      0.960686
7       0.976628
91      0.978000
137     0.990291
131     1.000000
33      1.000000
49      1.008902
98      1.009281
13      1.010568
22      1.018519
99      1.020408
138     1.030769
95      1.033333
2       1.036638
52      1.042945
104     1.054463
35      1.061693
96      1.062007
126     1.065089
156     1.067647
19      1.071429
34      1.074286
125     1.075964
158     1.078261
76      1.090909
67      1.128571
..           ...
71      1.877729
77      1.897805
30      1.910738
140     1.914286
44      1.917098
10      1.939163
72      1.950000
90      1.972973
106     1.973913
70      2.114286
134     2.122449
82      2.130984
109     2.176636
79      2.205403
20      2.211538
84      2.249166
143     2.261905
130     2.277916
55      2.280000
124     2.311561
64      2.319149
94      2.326531
108     2.371429
157     2.444444
63      2.681707
87      2.764706
127     2.969697
145     3.308271
120     4.218009
123  1115.384615

[159 rows x 1 columns]

<h3> É possível perceber que existem algumas porcentagens atuando como outliers para cima, como um aumento de 718875.81%, quanto para baixo, como um "aumento" de 0.000084%, ou seja, uma diminuição. Sendo assim será removido tanto os 5 maiores quanto os 5 menores valores  </h3>

In [23]:
for i in range(5):
    df_meanProfit = df_meanProfit[df_meanProfit['percentage'] != df_meanProfit['percentage'].max()]
    df_meanProfit = df_meanProfit[df_meanProfit['percentage'] != df_meanProfit['percentage'].min()]
df_meanProfit.sort_values(by=['percentage'])
#Aqui já é possível ver valores mais 'reais', para realização de uma boa análise

percentage
96     1.062007
126    1.065089
156    1.067647
19     1.071429
34     1.074286
125    1.075964
158    1.078261
76     1.090909
67     1.128571
110    1.144928
152    1.146043
149    1.166667
66     1.200000
147    1.205674
53     1.231084
28     1.238095
92     1.238095
16     1.280000
62     1.299750
51     1.311111
88     1.321867
65     1.342667
36     1.343636
50     1.348958
9      1.356618
117    1.366460
80     1.378151
25     1.382353
39     1.390604
12     1.404651
..          ...
151    1.708571
141    1.721944
119    1.722222
97     1.727564
103    1.731250
11     1.738088
54     1.738710
81     1.750000
132    1.757049
136    1.757143
73     1.758621
15     1.787440
57     1.794709
17     1.800554
32     1.805556
83     1.806452
56     1.807692
43     1.811594
112    1.837838
27     1.862857
71     1.877729
77     1.897805
30     1.910738
140    1.914286
44     1.917098
10     1.939163
72     1.950000
90     1.972973
106    1.973913
70     2.114286

[118 rows x 1 columns]

In [24]:
df_meanProfit.mean()

percentage    1.547665
dtype: float64

In [25]:
df_meanProfit.median()

percentage    1.57918
dtype: float64

In [26]:
df_meanProfit.mode()

percentage
0    1.238095

In [31]:
ax = sns.histplot((df_meanProfit-1)*100, bins = 70, kde = True)

ax.figure.set_size_inches(12, 6)
ax.set_title('Distribuição de Frequências - Acumulo dos valores dos calculos da valorização com o tempo', fontsize=18)
ax.set_xlabel('Porcentagem de valorização', fontsize=14)
ax

AttributeError: module 'seaborn' has no attribute 'histplot'

<h2> Pode-se inferir pelo resultado da média que as casas tendem a valorizar cerca de 25% com o tempo, porém se analisarmos a mediana, a moda e o gráfico acima, é possível observar que o maioria das casas tendem a desvalorizar com o tempo, uma vez que a mediana é menor que 1, ou seja, existe uma desvalorização na maioria das casas, o que também é possível de se observar no gráfico. Sendo assim, apenas esperar o tempo passar, não garantirá uma venda em um valor maior.
<br><br><br>
    Abaixo é explorado em qual mês o valor de venda é maior de acordo com os dados.
</h2>

In [ ]:
DataFrame['date'] = pd.to_datetime(DataFrame['date'])
for i in range(12):
    meanMonth = DataFrame[DataFrame['date'].dt.month == i+1]
    print('a média de valor no mês', i+1, 'é de:', format(meanMonth['price'].mean(),'.1E'))

In [ ]:
data_month = DataFrame
data_month['month'] = data_month['date'].dt.month


ax = sns.barplot(x = [i for i in range(1,13)], y =data_month.groupby('month').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o mês', fontsize=18)
ax.set_xlabel('Mês', fontsize=14)
ax.set_ylabel('Preço em Bilhões', fontsize=14)
ax

<h2> Vimos que o valor de vendas começa a aumentar em maio(5) atinge o auge em julho(7) e
em seguida os valores diminuem alcançando uma baixa em novembro(11), se igualando com vendas do primeiro semestre exceto abril(4).<br><br><br> Abaixo a análise a ser feita vai além dos meses mas também em qual ano foram feitas essas vendas.</h2>

In [ ]:
for j in range(2014, 2016, 1):
    for i in range(12):
        meanMonth = DataFrame[DataFrame['date'].dt.month == i+1]
        meanMonthPerYear = meanMonth[meanMonth['date'].dt.year == j]
        print('a média de valor no mês', i+1, 'no ano de', j,'é de:', format(meanMonthPerYear['price'].mean(),'.1E'))

In [ ]:
data_month = DataFrame
data_month['year-month'] = (data_month['date'].dt.year).astype(str) + '/' + (data_month['date'].dt.month).astype(str)

label = data_month['year-month'].value_counts().index


label 


In [ ]:
label = ['2015/04', '2014/07', '2014/06', '2014/08', '2014/10', '2015/03', '2014/09',
       '2014/05', '2014/12', '2014/11', '2015/02', '2015/01', '2015/05']
label = sorted(label)
values = data_month.groupby('year-month').price.mean()

values.index

In [ ]:
values.index = ['2014/10', '2014/11', '2014/12', '2014/05', '2014/06', '2014/07', '2014/08',
       '2014/09', '2015/01', '2015/02', '2015/03', '2015/04', '2015/05']

ax = sns.barplot(x = label , y = values.sort_index())
ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o mês e ano', fontsize=18)
ax.set_xlabel('Ano / Mês', fontsize=14)
ax.set_ylabel('Preço em Bilhões', fontsize=14)
ax

<h2> Nesta análise vê-se que os dados confirmam a ideia de que as vendas a partir de julho tendem a ser maiores e também é possível inferir uma valorização nas casas de 2014 para 2015, desta forma uma estratégia de negócio seria esperar a valorização do imóvel de um ano para o outro, se esta for uma opção viável.<br><br><br> A análise abaixo busca compreender o impacto do número de banheiros no valor da casa.</h2>

In [ ]:
bathrooms = DataFrame['bathrooms'].value_counts().sort_index()
save = bathrooms.tolist()
#print(bathrooms)
for i in range(len(bathrooms)):
    if i>3:
        meanVal = DataFrame[DataFrame['bathrooms'] == (i/4)]
        print('a média de valor para', i*0.25, 'banheiros é de:', format(meanVal['price'].mean(),'.1E'))
        save[i-3] = (meanVal['price'].mean())
    else:
        if i == 0:
            meanVal = DataFrame[DataFrame['bathrooms'] == i]
            print('a média de valor para', i, 'banheiros é de:', format(meanVal['price'].mean(),'.1E'))
            save[i] = (meanVal['price'].mean())
            
        elif i==1:
            meanVal = DataFrame[DataFrame['bathrooms'] == 7.5]
            print('a média de valor para', 7.5, 'banheiros é de:', format(meanVal['price'].mean(),'.1E'))
            save[27] = (meanVal['price'].mean())
            
        elif i==2:
            meanVal = DataFrame[DataFrame['bathrooms'] == 7.75]
            print('a média de valor para', 7.75, 'banheiros é de:', format(meanVal['price'].mean(),'.1E'))
            save[28] = (meanVal['price'].mean())

        elif i==3:
            meanVal = DataFrame[DataFrame['bathrooms'] == 8]
            print('a média de valor para', 8, 'banheiros é de:', format(meanVal['price'].mean(),'.1E'))
            save[29] = (meanVal['price'].mean())


In [ ]:
classes = [0, 0.99, 1.99, 2.99, 3.99, 4.99, 5.99, 6.99, 100]
labels = [0, 1, 2, 3, 4, 5, 6, '7+']
df_bathrooms = DataFrame
df_bathrooms['frequencia'] =  pd.cut(x = df_bathrooms.bathrooms,
         bins = classes,
         labels = labels,
         include_lowest = True)
df_bathrooms[df_bathrooms['frequencia'] == 1].price.mean()

In [ ]:
ax = sns.barplot(x = labels, y = df_bathrooms.groupby('frequencia').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de banheiros', fontsize=18)
ax.set_xlabel('Número de banheiros', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_bathrooms.groupby('frequencia').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Mediana dos preços das casas de acordo com o número de banheiros', fontsize=18)
ax.set_xlabel('Número de banheiros', fontsize=14)
ax.set_ylabel('Preço em 100 Milhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_bathrooms.groupby('frequencia').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Desvio padrão dos preços das casas de acordo com o número de banheiros', fontsize=18)
ax.set_xlabel('Número de banheiros', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> Pode-se observar que o número de banheiros provavelmente não é a variável que mais afeta o preço das casas, porém é interessante analisar que as casas da classe 6 banheiros tem uma valorização alta.</h2>

<h2> As análises abaixo serão feitas de mesma forma, mas levará em conta o número de quartos, andares, fontes de água frontal e view(que eu assumi como vista para pontos turísticos ou então paisagens).</h2>

In [ ]:
bedrooms = DataFrame['bedrooms'].value_counts().sort_index()
save = bedrooms.tolist()
#print(bedrooms)
for i in range(len(bedrooms)):
    if i == 12:
        meanVal = DataFrame[DataFrame['bedrooms'] == 33]
        print('a média de valor para', 33, 'quartos é de:', format(meanVal['price'].mean(),'.1E'))
        save[i] = (meanVal['price'].mean())
    else:    
        meanVal = DataFrame[DataFrame['bedrooms'] == i]
        print('a média de valor para', i, 'quarto(s) é de:', format(meanVal['price'].mean(),'.1E'))
        save[i] = (meanVal['price'].mean())

        
    
    

In [ ]:
classes = [0, 0.99, 1.99, 2.99, 3.99, 4.99, 5.99, 6.99, 100]
labels = [0, 1, 2, 3, 4, 5, 6, '7+']
df_bedrooms = DataFrame
df_bedrooms['frequencia'] =  pd.cut(x = df_bedrooms.bedrooms,
         bins = classes,
         labels = labels,
         include_lowest = True)
df_bathrooms[df_bathrooms['frequencia'] == 1].price.mean()

In [ ]:
ax = sns.barplot(x = labels, y = df_bedrooms.groupby('frequencia').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de quartos', fontsize=18)
ax.set_xlabel('Número de quartos', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_bedrooms.groupby('frequencia').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Mediana dos preços das casas de acordo com o número de quartos', fontsize=18)
ax.set_xlabel('Número de quartos', fontsize=14)
ax.set_ylabel('Preço', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_bedrooms.groupby('frequencia').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Desvio padrão dos preços das casas de acordo com o número de quartos', fontsize=18)
ax.set_xlabel('Número de quartos', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> Pela média é possível concluir que ter quartos é uma decisão ruim, o que me faz pensar que estes dados não são apenas de casas e sim de qualquer construção, incluindo edifícios para negócios. Entretanto, observando a mediana e desvio padrão podemos perceber que o aumento de quartos tende a aumentar a mediana e o desvio padrão em comparação com a média, sendo assim podemos concluir que o número de quartos influência o valor de uma casa, não de um edifício de negócio onde não existe a necessidade de um quarto. Sendo assim percebo que para analisar melhor a base de dados seria necessário separar as casas das demais construções.</h2>

In [ ]:
floors = DataFrame['floors'].value_counts().sort_index()
save = floors.tolist()
#print(floors)
for i in range(len(floors)):
    meanVal = DataFrame[DataFrame['floors'] == ((i*0.5)+1)]
    print('a média de valor para', ((i*0.5)+1), 'andar(es) é de:', format(meanVal['price'].mean(),'.1E'))
    save[i] = (meanVal['price'].mean())


In [ ]:
classes = [1, 1.99, 2.99, 100]
labels = [ 1, 2, '3+']
df_floors = DataFrame
df_floors['frequencia'] =  pd.cut(x = df_floors.floors,
         bins = classes,
         labels = labels,
         include_lowest = True)

In [ ]:
ax = sns.barplot(x = labels, y = df_floors.groupby('frequencia').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de andares', fontsize=18)
ax.set_xlabel('Número de andares', fontsize=14)
ax.set_ylabel('Preço em Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_floors.groupby('frequencia').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Mediana dos preços das casas de acordo com o número de andares', fontsize=18)
ax.set_xlabel('Número de andares', fontsize=14)
ax.set_ylabel('Preço', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_floors.groupby('frequencia').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Desvio Padrão dos preços das casas de acordo com o número de andares', fontsize=18)
ax.set_xlabel('Número de andares', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> Assim como nos dados sobre banheiros, nos dados de andares vemos que o ideal do imóvel é ter 2 andares, pois este se mantém em alta tanto na média quanto na mediana, porém como seu desvio padrão também é elevado, podemos concluir que não é somente a quantidade de andares que afetará o preço da casa ou edifício.</h2>

In [ ]:
waterfront = DataFrame['waterfront'].value_counts().sort_index()
save = waterfront.tolist()
#print(waterfront)
for i in range(len(waterfront)):
    meanVal = DataFrame[DataFrame['waterfront'] == i]
    print('a média de valor para', i, 'fonte de água frontal é de:', format(meanVal['price'].mean(),'.1E'))
    save[i] = (meanVal['price'].mean())

In [ ]:
ax = sns.barplot(x = [0,1], y = df_floors.groupby('waterfront').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de fontes de água frontal', fontsize=18)
ax.set_xlabel('Número de fontes de água frontal', fontsize=14)
ax.set_ylabel('Preço em Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = [0,1], y = df_floors.groupby('waterfront').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de fontes de água frontal', fontsize=18)
ax.set_xlabel('Número de fontes de água frontal', fontsize=14)
ax.set_ylabel('Preço em Milhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = [0,1], y = df_floors.groupby('waterfront').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de fontes de água frontal', fontsize=18)
ax.set_xlabel('Número de fontes de água frontal', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> Sobre os resultados acima, vemos que com uma mediana na casa dos milhões e com o desvio padrão e média na casa dos bilhões podemos concluir que a fonte pode até influênciar o valor, mas certamente não é um fator principal neste quesito.</h2>

In [ ]:
view = DataFrame['view'].value_counts().sort_index()
save = view.tolist()
#print(view)
for i in range(len(view)):
    meanVal = DataFrame[DataFrame['view'] == i]
    print('a média de valor para', i, 'vista(s) é de:', format(meanVal['price'].mean(),'.1E'))
    save[i] = (meanVal['price'].mean())

In [ ]:
classes = [0, 0.99, 1.99, 2.99, 3.99, 100]
labels = [ 0, 1, 2, 3, '4+']
df_view = DataFrame
df_view['frequencia'] =  pd.cut(x = df_view.view,
         bins = classes,
         labels = labels,
         include_lowest = True)

In [ ]:
ax = sns.barplot(x = labels, y = df_view.groupby('frequencia').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_view.groupby('frequencia').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em Milhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_view.groupby('frequencia').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> Pelos dados acima pode-se dizer que é interessante para a empresa observar quais casas possuem uma(1) vista no mínimo e de preferência 4 vistas. Isso se dá por conta da média e mediana crescente de acordo com o n° de vistas, porém o desvio padrão é muito alto. Sendo assim, a vista afeta o preço, porém não é garantia de uma valorização gigantesca na casa.</h2>


<br>
<br>
<br>

<h2>Respondendo as perguntas do início:</h2>

In [ ]:
classes = [0, 2000, 5000, 10000, 12000, 100000]
labels = [ '0-2000', '2000-5000', '5000-10000', '10000-12000', '12000+']

df_sqf = DataFrame
df_sqf['frequencia'] = pd.cut(x = df_sqf.sqft_living,
         bins = classes,
         labels = labels,
         include_lowest = True)

In [ ]:
ax = sns.barplot(x = labels, y = df_sqf.groupby('frequencia').price.mean())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_sqf.groupby('frequencia').price.median())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

In [ ]:
ax = sns.barplot(x = labels, y = df_sqf.groupby('frequencia').price.std())

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h2> 1. Quais casas o CEO da House Rocket deveria comprar e por qual preço de compra? </h2>

<h3>Em resumo as casas que o CEO da House Rocket deveria comprar são:
    <ol>
        <li>Casas com valores menores que 200 mil</li>
        <li>De preferência casas passíveis de serem reformadas, ou seja, que não tenham sido reformadas.</li>
        <li>De preferência casas no ZipCode 98039.</li>
        <li>Casas de 3 a 5 banheiros, e caso não tenha ao ser comprado é indicada a construção dos mesmos na reforma.</li>
        <li>Casas de 2 a 5 quartos, e caso não tenha ao ser comprado é indicada a construção dos mesmos na reforma.</li>
        <li>Casas de 2 andares, e caso não tenha ao ser comprado é indicada a construção dos mesmos na reforma.</li>
        <li>Casas com 1 fonte de água frontal, e caso não tenha ao ser comprado é indicada a construção da mesma na reforma.</li>
        <li>Casas com vista e de preferência com o valor igual a 4.</li>
    </ol>
</h3>

In [ ]:
wantedHouses = DataFrame.iloc[0:0]

#1
df1 = DataFrame[DataFrame['price'] < 200000]

#2
#df1 = df1[df1['zipcode'] == 98039]

#4

wantedHouses = wantedHouses.append(df1[df1['bathrooms'] <= 5])

#5
df1 = wantedHouses 
wantedHouses = wantedHouses.loc[wantedHouses['bedrooms'] <= 5]

#6
df1 = wantedHouses 
wantedHouses = wantedHouses.loc[wantedHouses['floors'] <= 2]

#7 não há casas que tenham fonte
wantedHouses = wantedHouses.loc[wantedHouses['waterfront'] <= 1]

#8

wantedHouses = wantedHouses[wantedHouses['view'] > 0]

wantedHouses.sort_values(by = 'price')

In [ ]:
DataFrame['sqft_lot'].mean()

<h3> Essas seriam as 5 melhores opções. Dado as condições possíveis, uma vez que no zipcode preferencial não estão dentro da faixa de preço.
    <br><br>
A melhor opção seria o imóvel de id igual a 1320069249 com o valor de 192mil, pois é uma casa que atende os requisitos e dado ao seu grande espaço de terreno após uma reforma a empresa conseguiria ter um lucro com a casa, uma vez que uma casa renovada tem a expectativa de aumento, além das outras melhorias possíveis de novos quartos, banheiros, andares e uma fonte de água também podem aumentar o lucro, e por fim ainda sobrar espaço no terreno. 
    <br><br>
    Tudo isso levado em conta, é possível que a divisão deste terreno seja utilizado para construção de novas casas, pois o terreno deste imóvel é enorme, dado por "sqft_lot" = 63737, sendo 4 vezes o tamanho médio dos imóveis, desta forma além de lucrar com a reforma na casa já existente é possível lucrar com novos imóveis construídos.</h3>

<h2> 2. Uma vez a casa em posse da empresa, qual o melhor momento para vendê-las e qual seria o preço da venda? </h2>

<h3>
    Caso a casa passe após a reforma de 1 quarto para 3 quartos, 1 banheiro para 3 banheiros, 2 andares e uma fonte de água na frente da casa. É possível que o valor desta casa triplique facilmente, uma vez que os dados informam que essas alterações seriam suficientes para aumentar o valor da casa em mais de 10 vezes, porém sabemos que somente essas alterações não seriam suficientes, seria necessário um maior conhecimento do mercado local, analisar seu posicionamento (zipcode) entre outros fatores que não é possível ser analisado pelos dados. Desta forma a afirmação de que o valor possa triplicar não foge à realidade dos dados e nem à realidade dos negócios, podendo assim dar um "norte" para empresa sobre um possível retorno na venda da casa pós reforma.
    
</h3>

<h2> 3. A House Rocket deveria fazer uma reforma para aumentar o preço da venda? Quais seriam as sugestões de mudanças? Qual o incremento no preço dado por cada opção de reforma? </h2>

<h3>
    <ol>
        <li> A House Rocket deveria fazer uma reforma para aumentar o preço da venda?</li>
            <p style="font-size: 18px;">-Sim, seria recomendado pois como foi analisado é esperado uma valorização após a reforma, baseado na média de valores das casas reformadas e das não reformadas.</p><br>
        <li> Quais seriam as sugestões de mudanças?</li>
        <p>Como foi visto anteriormente os requisitos para uma casa com um bom valor de mercado são os seguintes:</p>
        <p>
        * 3 a 5 banheiros.<br>
        * 2 a 5 quartos.<br>
        * 2 andares.<br>
        * 1 fonte de água frontal.<br>
        * vista e de preferência com o valor igual a 4.</p><br>
        <p> Os dados indicam um aumento muito absurdo, exemplo: o preço médio de casas de um banheiro para casas com 5 banheiros é de 100 vezes o preço. <br><br> A análise que pode ser feita é de que existe um aumento no valor da casa se esta possuir mais recursos, como mais banheiros, quartos, etc. Sendo assim, pode-se dizer que os valores citados acima seriam encontrados em casas com uma maior valorização, mas que não necessáriamente, você ter 5 banheiros em uma casa esta terá sua valorização multiplicada por 100.</p><br>
        <li> Qual o incremento no preço dado por cada opção de reforma?</li>
        <p>Essa informação está muito enviesada nos dados, uma vez que não temos uma mesma casa (mesmo ID) que tenha construído mais um banheiro ou um quarto. Sendo assim, seria necessárias casas que tenham havido mudanças em sua construção, por exemplo, uma casa que tenha construído mais um banheiro, para saber o impacto do mesmo no preço, isso acontecendo diversas vezes poderíamos calcular um valor de valorização mais exato.<br>
Com os dados atuais a única afirmação possível de ser feita é que dado uma casa com os requisitos citados acima, o esperado é que esta tenha um alto valor em relação a casas que não cumprem esses requisitos.
</p><br>
    </ol>
    
</h3>

   

In [ ]:
ax = sns.boxplot(y = DataFrame.price)

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

In [ ]:
ax = sns.boxplot(y = DataFrame.query('price < 1000000').price)

ax.figure.set_size_inches(12, 6)
ax.set_title('Média dos preços das casas de acordo com o número de "vista"', fontsize=18)
ax.set_xlabel('Número de "vista"', fontsize=14)
ax.set_ylabel('Preço em 10 Bilhões', fontsize=14)

<h1>Dado os boxplots, é notada a presença de diversos outliers que distorcem as informações. Sendo assim, o projeto será refatorado dividindo as bases em duas bases de dados, Casas e Casas de Luxo.</h1>